fem/bilininteg.hpp

```diff
@@ -2671,11 +2671,22 @@ class ElasticityIntegrator : public BilinearFormIntegrator
 /** Integrator for the DG form:
     alpha < rho_u (u.n) {v},[w] > + beta < rho_u |u.n| [v],[w] >,
     where v and w are the trial and test variables, respectively, and rho/u are
-    given scalar/vector coefficients. The vector coefficient, u, is assumed to
-    be continuous across the faces and when given the scalar coefficient, rho,
-    is assumed to be discontinuous. The integrator uses the upwind value of rho,
-    rho_u, which is value from the side into which the vector coefficient, u,
-    points. */
+    given scalar/vector coefficients. {v} represents the average value of v on
+    the face and [v] is the jump such that {v}=(v1+v2)/2 and [v]=(v1-v2) for the
+    face between elements 1 and 2. For boundary elements, v2=0. The vector
+    coefficient, u, is assumed to be continuous across the faces and when given
+    the scalar coefficient, rho, is assumed to be discontinuous. The integrator
+    uses the upwind value of rho, rho_u, which is value from the side into which
+    the vector coefficient, u, points.
+
+    When used with ConvectionIntegrator, the transpose of this form with
+    alpha=-1.0, beta=0.5 can be used to implement the upwind flux and outflow
+    boundary conditions, see ex9 and ex9p.
```

Comment:
```suggestion
    One use case for this integrator is to discretize the operator -u.grad(v)
    with a DG formulation. The resulting formulation uses the
    ConvectionIntegrator (with coefficient u, and parameter alpha = -1) and the
    transpose of the DGTraceIntegrator (with coefficient u, and parameters
    alpha = 1, beta = -1/2 to use the upwind face flux). This discretization and
    the handling of the inflow boundary conditions is illustrated in Example
    9/9p.
```
I'm not sure why the current comment talks about outflow b.c. In Example 9/9p there are b.c. _only_ on the inflow boundary. Am I missing something?

Reply:
The inflow boundaries are handled by `BoundaryFlowIntegrator` on the RHS. Outflow boundaries in ex9 are handled by the `DGTraceIntegrator` (perhaps it's not quite correct to use the term "outflow boundary conditions" — no BCs are enforced at outflow boundaries, this form handles the correct upwinding).

Reply:
I think the outflow b.c. is the correct b.c. when discretizing `u.grad(v)` -- I think this is what the current comment is describing. I used `-u.grad(v)` because that is the r.h.s. operator in the time-dependent operator in example 9/9p: `dv/dt = -u.grad(v)`.